syntax = "proto3";

package auraservice;

service AdaptiveAuraService {
  rpc AdaptiveExecute (ExecutionRequest) returns (ExecutionResponse);
  rpc SelfOptimize (OptimizationRequest) returns (OptimizationResponse);
  rpc GetMetrics (MetricsRequest) returns (MetricsResponse);
}

message ExecutionRequest {
  string method_name = 1;
  map<string, string> args_json = 2;
}

message ExecutionResponse {
  string result_json = 1;
  string status = 2;
  string optimization_notes = 3;
}

message OptimizationRequest {
  string target = 1;
}

message OptimizationResponse {
  string message = 1;
  string original_hash = 2;
  string optimized_hash = 3;
}

message MetricsRequest {}

message MetricsResponse {
  string metrics_json = 1;
}
